birdhouse/deprecated-components/README.rst

```diff
@@ -0,0 +1,29 @@
+Deprecated Components
+#####################
+
+.. contents::
+
+All components in this directory are not actively maintained.
+
+If you wish to include these components in the deployed stack, additional work
+may be required to make them compatible with the current stack.
+
+Components in this directory may be removed from this repository at any time.
+
+This directory also contains additional configurations for these deprecated components in:
+
+- `all-public-access-deprecated/`
+- `canarie-api-full-monitoring-deprecated/`
+- `wps-healthchecks-deprecated/`
```

Comment:
Thinking about this again... Is there actually any reason to move them under a separate configuration?

If I activated for example: 

```shell
EXTRA_CONF_DIRS="
  ./components/weaver
  ./deprecated-components/malleefowl
  ./optional-components/all-public-access-deprecated
  ./optional-components/canarie-api-full-monitoring
  ./optional-components/wps-healthchecks
"
```

Wouldn't the following be loaded (not necessarily exhaustive or in the right order below)?
```shell
./components/weaver/*
./deprecated-components/malleefowl
./optional-components/all-public-access-deprecated/mallefowl/*
./optional-components/all-public-access-deprecated/weaver/*
./optional-components/canarie-api-full-monitoring/mallefowl/*
./optional-components/canarie-api-full-monitoring/weaver/*
./optional-components/wps-healthchecks/mallefowl/*
./optional-components/wps-healthchecks/weaver/*
```

In order words, whether the main services are under `config`, `components` or `deprecated-components`, the `optional-components` to use would load whichever service name that it matches nested under its directory.

I find this would be easier to maintain because a service moving between `config`, `components` or `deprecated-components` states would not require updating the `optional-components`, nor require loading another `(optional+deprecated)-component` variant equivalent to the original one.

Reply:
> Wouldn't the following be loaded (not necessarily exhaustive or in the right order below)? 

At the moment yes but that is a mistake. I just forgot to delete the catalog and malleefowl directories under optional-components/wps-healthchecks/

I've remove them now.

> I find this would be easier to maintain because a service moving between config, components or deprecated-components states would not require updating the optional-components, nor require loading another (optional+deprecated)-component variant equivalent to the original one.

I think that you need to think about whether you actually want to deprecate these components or not. In my opinion, the point of moving them into a separate folder is to clearly mark that they are deprecated and no longer maintained. Developers then can focus on maintaining the other components without worrying about breaking components that we don't care about anymore. 

The only reason I can think of to keep supporting code in the optional-components folder is if you think that these components will be "un-deprecated" at some point.

Reply:
If someone needs to work on, for example, `optional-components/wps-healthchecks` to add a new WPS service, this doesn't require them to update all nested components. The `optional-components/wps-healthchecks/malleefowl` for example would be left untouched. I don't think leaving them under `optional-components` will cause any breaking issue, nor would affect maintenance burden.

Reply:
Up to you... I moved them back